src/enclave_proc/resource_manager.rs

```diff
@@ -469,13 +470,26 @@ impl EnclaveHandle {
         debug_mode: bool,
     ) -> NitroCliResult<Self> {
         let requested_mem = memory_mib << 20;
+        let link_path_res = read_link(PathBuf::from(format!(
+            "/proc/self/fd/{}",
+            eif_file.as_raw_fd()
+        )));
+        let link_path;
+        let mut eif_path = "Unknown file";
+        if link_path_res.is_ok() {
+            link_path = link_path_res.unwrap();
+            eif_path = link_path
+                .to_str()
+                .unwrap_or("Invalid unicode EIF file name");
+        }
```

Comment:
I don't think this is necessary. In such cases it would fail when you open the file, not here.

Reply:
I know, but the error related to invalid EIF name comes from the conversion from `std::fs::Path` to `str` implied by `link_path.to_str()` - in order to avoid `unwrap()`ping the result, I think the error case should be considered explicitly here.